src/cpp/session/modules/build/SessionInstallRtools.cpp

```diff
@@ -89,6 +92,31 @@ Error installRtools()
    if (error)
       return error;
 
+   if (version == "4.2")
+   {
+      try
+      {
+         FilePath tmpFile;
+         FilePath::tempFilePath(tmpFile);
+         r::exec::RFunction("utils:::download.file")
+               .addParam("url", url)
+               .addParam("destfile", tmpFile.getAbsolutePath())
+               .addParam("mode", "rb")
+               .call();
+
+         std::string rToolsHome = file_utils::readFile(tmpFile);
+         boost::regex reLinkPattern("<a\\shref=\"(.*\\.exe)\">\\s*Rtools42 installer<\\/a>", boost::regex::icase);
```

Comment:
I wonder if we should be more permissive? I worry a little that the `Rtools42 installer` text could change; it'd be easier if we could just iterate through all of the `<a href ="<...>">` entries in the document and pick the one that "looks like" the Rtools installer.

For reference, if you want to find and iterate over multiple regular expression matches, you could use a regex_token_iterator. See:

https://github.com/rstudio/rstudio/blob/8be41624a8de5d4caf60ab6510db57c3a22000c4/src/cpp/session/modules/rmarkdown/SessionBlogdown.cpp#L238-L252

for a somewhat similar example.

Alternatively, you could move all of this logic into R and have a helper function like `.rs.findRtools42Installer()`